internal/provider/common_cert.go

```diff
@@ -122,10 +122,14 @@ func setCertificateSubjectSchema(s map[string]*schema.Schema) {
 	}
 
 	s["key_algorithm"] = &schema.Schema{
-		Type:        schema.TypeString,
-		Required:    true,
-		ForceNew:    true,
-		Description: "Name of the algorithm used when generating the private key provided in `private_key_pem`.",
+		Type:     schema.TypeString,
+		Optional: true,
+		ForceNew: true,
+		Deprecated: "This is now ignored, as the key algorithm is inferred from the `private_key_pem`. " +
+			"It it will be made read-only in the next major release.",
```

Comment:
In my testing I concluded that a 2 step approach (first make it `Optional`, release, then make it `Computed`) was necessary to avoid causing unnecessary re-creation of the resources.

I can remove the reference to "future behaviour" as what matters is that ppl stop using it.

Reply:
I should've clarified better, sorry. I meant `Optional` and `Computed`, with calling `d.Set()`. `Optional` + `Computed` + `ForceNew` should not trigger recreation when removing the attribute from the configuration and if it does, that would be unexpected behavior we should investigate further.

Reply:
So, here is the issue with adding `Computer` to `Optional` now: it will be a breaking change.

I think we are expected to cross that bridge only once we release a major version, after a minor version with the deprecation - no?

Reply:
To be clear, I tried yesterday all the combinations, and just changing from `Required` to `Optional` for now, seemed like the only path forward to allow for an opt-in migration.

All other options would end up with a breaking change.